Gems/Atom/RHI/Vulkan/Code/Source/RHI/FunctionLoader.cpp

```diff
@@ -0,0 +1,111 @@
+                            context->CreateDebugUtilsMessengerEXT = reinterpret_cast<PFN_vkCreateDebugUtilsMessengerEXT>(
+                                context->GetInstanceProcAddr(vkInstance, "vkCreateDebugUtilsMessengerEXT"));
+                            context->DestroyDebugUtilsMessengerEXT = reinterpret_cast<PFN_vkDestroyDebugUtilsMessengerEXT>(
+                                context->GetInstanceProcAddr(vkInstance, "vkDestroyDebugUtilsMessengerEXT"));
+                            context->SubmitDebugUtilsMessageEXT = reinterpret_cast<PFN_vkSubmitDebugUtilsMessageEXT>(
+                                context->GetInstanceProcAddr(vkInstance, "vkSubmitDebugUtilsMessageEXT"));
+                        }
+
+                        if (device != VK_NULL_HANDLE)
+                        {
+                            context->CmdBeginDebugUtilsLabelEXT = reinterpret_cast<PFN_vkCmdBeginDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkCmdBeginDebugUtilsLabelEXT"));
+                            context->CmdEndDebugUtilsLabelEXT = reinterpret_cast<PFN_vkCmdEndDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkCmdEndDebugUtilsLabelEXT"));
+                            context->CmdInsertDebugUtilsLabelEXT = reinterpret_cast<PFN_vkCmdInsertDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkCmdInsertDebugUtilsLabelEXT"));
+                            context->QueueBeginDebugUtilsLabelEXT = reinterpret_cast<PFN_vkQueueBeginDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkQueueBeginDebugUtilsLabelEXT"));
+                            context->QueueEndDebugUtilsLabelEXT = reinterpret_cast<PFN_vkQueueEndDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkQueueEndDebugUtilsLabelEXT"));
+                            context->QueueInsertDebugUtilsLabelEXT = reinterpret_cast<PFN_vkQueueInsertDebugUtilsLabelEXT>(
+                                context->GetDeviceProcAddr(device, "vkQueueInsertDebugUtilsLabelEXT"));
+                            context->SetDebugUtilsObjectNameEXT = reinterpret_cast<PFN_vkSetDebugUtilsObjectNameEXT>(
+                                context->GetDeviceProcAddr(device, "vkSetDebugUtilsObjectNameEXT"));
+                            context->SetDebugUtilsObjectTagEXT = reinterpret_cast<PFN_vkSetDebugUtilsObjectTagEXT>(
+                                context->GetDeviceProcAddr(device, "vkSetDebugUtilsObjectTagEXT"));
+                        }
+                    }
+                }
+            }
+        }
+
+        bool FunctionLoader::LoadProcAddresses(GladVulkanContext* context, VkPhysicalDevice physicalDevice, VkDevice device)
+        {
+            VkInstance instance = Instance::GetInstance().GetNativeInstance();
+            int result = gladLoaderLoadVulkanContext(context, instance, physicalDevice, device);
```

Comment:
I actually prefer the previous function declaration which have VkInstance as input parameter. 
It's clear that people notice the different of the three LoadProcAddresses calls and won't wonder why there are two LoadProcAddresses calls with same parameters.